datahub-web-react/src/app/identity/user/UserListItem.tsx

```diff
@@ -45,12 +66,17 @@ const MenuIcon = styled(MoreOutlined)<{ fontSize?: number }>`
     margin-left: 5px;
 `;
 
-export default function UserListItem({ user, canManageUserCredentials, onDelete }: Props) {
+export default function UserListItem({ user, canManageUserCredentials, roles, onDelete, refetch }: Props) {
     const entityRegistry = useEntityRegistry();
     const [isViewingResetToken, setIsViewingResetToken] = useState(false);
+    const [isViewingAssignRole, setIsViewingAssignRole] = useState(false);
+    const [roleToAssign, setRoleToAssign] = useState<DataHubRole>();
     const displayName = entityRegistry.getDisplayName(EntityType.CorpUser, user);
     const isNativeUser: boolean = user.isNativeUser as boolean;
     const shouldShowPasswordReset: boolean = canManageUserCredentials && isNativeUser;
+    const userRelationships = user.relationships?.relationships;
+    const userRole = userRelationships && userRelationships.length > 0 && (userRelationships[0]?.entity as DataHubRole);
```

Comment:
I'm slightly confused by this - why are we finding the userRole inside userRelationships?

Reply:
oh are you saying we should just resolve the role on the graphql side and not in relationships? this is the same pattern as groups. This implementation also might be confusing since right now we're enforcing the invariant of a single role but it could be a list in the future

Reply:
Yeah I feel that since we have RoleMembership we should have a roleMembershiip field on the user in the long run